email/add_ecr/email_sender.py

```diff
@@ -0,0 +1,311 @@
+        html_cards = html_cards+generate_author_html_container(id)
+
+    html = f'''
+    <body style="font-family: 'Poppins', Arial, sans-serif; background-color: #f9f9f9; padding: 20px; color: #333333; line-height: 1.5;>
+      <table width="100%" cellpadding="0" cellspacing="0" border="0" style="max-width: 600px; margin: 0 auto; background-color: #ffffff; border: 1px solid #e0e0e0; padding: 20px;">
+        <tr>
+          <td>
+            <p style="font-size: 16px; margin: 0 0 15px 0;">Dear <strong>{publisher_name}</strong>,</p>
+            <p style="font-size: 15px; margin: 0 0 20px 0;">The following authors had noteworthy engagement:</p>
+      <table width="100%" border="0" cellspacing="0" cellpadding="0" style="background-color: #fafafa; border-radius: 6px; overflow: hidden;">
+    '''
+    html = html + html_cards
+    return html + '</body>'
+
+
+def format_html_email(head: str, body: str) -> str:
+    '''Formats the html for an update email'''
+    return '<html lang="en">' + head + body + '</html>'
+
+
+def get_email_subject(publisher_id: int) -> str:
+    '''Formats and returns the email subject for a given publisher id'''
+    date = get_todays_date_formatted()
+    name = get_publishers_name(publisher_id)
+    return f'{name}\'s Daily Tracker {date}'
+
+
+def aws_send_email(email_html: str, publisher_id: int):
+    """Send the specified email HTML using AWS SES."""
+    client = boto3.client("ses",
+                          region_name="eu-west-2",
+                          aws_access_key_id=os.getenv('ACCESS_KEY'),
+                          aws_secret_access_key=os.getenv('SECRET_KEY'))
+    message = MIMEMultipart()
+    message["Subject"] = get_email_subject(publisher_id)
+    message["From"] = "trainee.rodrigo.montemayor@sigmalabs.co.uk"
+    message["To"] = get_publishers_email(publisher_id)
+
+    body = MIMEText(email_html, 'html')
+    message.attach(body)
+
+    client.send_raw_email(
+        Source=message["From"],
+        Destinations=[message["To"]],
+        RawMessage={"Data": message.as_string()}
+    )
+
+
+def send_email_to_publisher(publisher_id: int) -> None:
+    '''Sends the update email to the specified author.'''
+    html_body = generate_html_body(publisher_id)
+    html_head = generate_html_head()
+    html_email = format_html_email(html_head, html_body)
+    aws_send_email(html_email, publisher_id)
+
+
+def send_email_to_all_publishers() -> None:
+    '''Sends an email containing information about their tracked authors
+    to all publishers of the bookworm dashboard in our database.'''
+    publisher_ids = get_publisher_ids()
+    for id in publisher_ids:
+        print(id)
+        try:
+            send_email_to_publisher(id[0])
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'MessageRejected':
+                print("Message was rejected:", e.response['Error']['Message'])
+            else:
+                raise e
+
+
+def lambda_handler(event, context):
```

Comment:
Missing type hints here for the return value and arguments

```suggestion
def lambda_handler(event: None = None, context = None ) -> None:
```